src/components/input/InputGroup.tsx

```diff
@@ -12,7 +12,9 @@ export const inputGroupStyles = classnames(
   // Restore border-radius on the leftmost and rightmost components in the group
   'input-group:first:rounded-l input-group:last:rounded-r',
   // "Collapse" borders between input components
-  'input-group:border-l-0 input-group:first:border-l',
+  'input-group:ml-[-1px] input-group:first:ml-0',
+  // Make sure focused element appears on top, preventing a cropped focus ring
+  'focus:z-1',
```

Comment:
This ensures the active/focused element will render on top of the other items of the group, so that its borders and shadows are visible.

Reply:
Example of cropped ring without this change:

<img width="455" alt="Cropped focus ring" src="https://github.com/user-attachments/assets/b37f3dbb-c445-4660-83d5-cb6b169586ba">